### PR TITLE
Veronika algolia plugin

### PR DESCRIPTION
**Pull request by VPilipenko334**
This is an Algolia Search Integration to add to the unified-demo. This way, if customers are planning on using Algolia, we can demo how it integrates with Builder, as well as be able to show them how it works in the Visual Editor UI. 

[Here](https://www.loom.com/share/fa6084f3e59347d0a35069a78f235fe1?sid=d2f3c303-9b9e-428b-b1fe-7cfedfb40b05) is a loom video walkthrough of me using it.

Screenshot 2024-11-01 at 4 46 47 PM

### CODE REVIEW

**Comment by VPilipenko334** [comment]
maybe kind of a silly question but do you know why GitGuardian is saying that the API keys are exposed? I was under the impression that . meant it would omit that file from being uploaded/shown

**Comment by tgaribal** [comment]
@VPilipenko334  looks great! Could you maybe make this its own custom component like AlgoliaSearchBox instead of having it in the ProductCard component? And then in the custom insertMenu config in builder registery lets add it to the Layout section

File: ecommerce/src/components/AlgoliaSearch/AlgoliaSearch.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    </div>
+  );
+}
+
+function Hit({ hit }: { hit: HitData }) {
+  if (!hit || Object.keys(hit).length === 0) {
+    return null;
+  }
+  return (
+    <div className="hit-container w-64 border p-4 mb-4 flex flex-col items-center">
```
Comment (tgaribal) [inline ecommerce/src/components/AlgoliaSearch/AlgoliaSearch.tsx:69]: @VPilipenko334  also can we use the ProductCard component here to match the other styles?  the algolia data is based on our product data model I believe so should be able to just pop in the product

**Comment by VPilipenko334** [comment]
[Here](https://www.loom.com/share/52050d1b5a9f47978a7af0b23ccbdc50?sid=b9b24d3b-cf5d-4f56-bb4c-e4e787f04fa1) is a new loom video of the changes implemented!

**Pull request merged**